src/rules/utils.ts

```diff
@@ -758,3 +758,32 @@ export const scopeHasLocalReference = (
     !references.unresolved.has(referenceName)
   );
 };
+
+/**
+ * Checks if node names returned by getNodeName matches any of the given star patterns
+ * Pattern examples:
+ *   request.*.expect
+ *   request.**.expect
+ *   request.**.expect*
+ */
+
+export function nodeNameMatch(
+  nodeName: string,
+  ...patterns: readonly string[]
+): boolean {
+  for (const p of patterns) {
```

Comment:
I think this'd look cleaner if it was using functional array methods - specifically `#some`.

That would save you having to use the if condition, and (combined with an arrow function) explicit returns.